developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -47,33 +47,89 @@ const baseOptions: CompilerOptions = {
  */
 let callbackProcIdentifier = 0;
 
+/**
+ * Pointer in wasm-space
+ */
+type WasmPtr = number;
+
+/**
+ * The wrapped functions
+ */
+class WasmWrapper {
+  Module: any;
+
+  compileKeyboardFile?: (pszInfile: string, pszOutfile: string, aSaveDebug: number, aCompilerWarningsAsErrors: number, aWarnDeprecatedCode: number, msgProc: string) => boolean;
+  parseUnicodeSet?: (pat: string, buf: WasmPtr, length: number) => number;
+  setCompilerOptions?: (shouldAddCompilerVersion: number) => boolean;
+
+  constructor(wasmModule: any) {
+    this.Module = wasmModule;
+    if (!wasmModule) {
+      throw Error(`wasm host did not load`);
+    }
+    this.compileKeyboardFile = this.Module.cwrap('kmcmp_Wasm_CompileKeyboardFile', 'boolean', ['string', 'string', 'number', 'number', 'number', 'string']);
+    this.parseUnicodeSet = this.Module.cwrap('kmcmp_Wasm_ParseUnicodeSet', 'number', ['string', 'number', 'number']);
+    this.setCompilerOptions = this.Module.cwrap('kmcmp_Wasm_SetCompilerOptions', 'boolean', ['number']);
+
+    if (this.parseUnicodeSet == undefined
+      || this.setCompilerOptions == undefined
+      || this.compileKeyboardFile == undefined) {
+        throw Error(`some wasm functions did not load properly.`);
+      }
+  }
+
+  /**
+   * Entry point into Wasm functions
+   * @returns WasmWrapper
+   */
+  public static async load() : Promise<WasmWrapper> {
+    return new WasmWrapper(await loadWasmHost());
+  }
+};
+
 export class Compiler {
-  wasmModule: any;
-  compileKeyboardFile: any;
-  setCompilerOptions: any;
   callbackName: string;
   callbacks: CompilerCallbacks;
+  wasm: WasmWrapper;
 
   constructor() {
     this.callbackName = 'kmnCompilerCallback' + callbackProcIdentifier;
     callbackProcIdentifier++;
   }
 
-  public async init(): Promise<boolean> {
-    if(!this.wasmModule) {
-      this.wasmModule = await loadWasmHost();
-      this.compileKeyboardFile = this.wasmModule.cwrap('kmcmp_Wasm_CompileKeyboardFile', 'boolean', ['string', 'string',
-        'number', 'number', 'number', 'string']);
-      this.setCompilerOptions = this.wasmModule.cwrap('kmcmp_Wasm_SetCompilerOptions', 'boolean', ['number']);
+  public async init(callbacks: CompilerCallbacks): Promise<boolean> {
+    if(!this.callbacks) {
+      this.callbacks = callbacks;
     }
```

Comment:
Why not overwrite with parameter in all cases? Seems like it may be confusing otherwise.

```suggestion
    this.callbacks = callbacks;
```

Reply:
makes sense.